modules/tinymce/src/themes/silver/main/ts/ui/core/complex/BespokeNumberInput.ts

```diff
@@ -0,0 +1,185 @@
+              return Optional.some(true);
+            },
+            onEscape: goToParent,
+            onUp: (_comp) => {
+              increase(false);
+              // TOFIX: now it preserve the focus but it put the selection at the end of the input
+              return Optional.some(true);
+            },
+            onDown: (_comp) => {
+              decrease(false);
+              return Optional.some(true);
+            }
+          })
+        ])
+      })
+    ],
+    behaviours: Behaviour.derive([
+      Focusing.config({}),
+      Keying.config({
+        mode: 'special',
+        onEnter: (comp) => {
+          if (Focus.hasFocus(comp.element)) {
+            Traverse.firstChild(comp.element).each((input) => Focus.focus(input as SugarElement<HTMLElement>));
+            return Optional.some(true);
+          } else {
+            return Optional.none();
+          }
+        },
+        onEscape: goToParent
+      })
+    ])
+  });
+
+  const moveFocus = (parentComp: AlloyComponent, direction: 'left' | 'right'): void => {
+    const nextNode = direction === 'right' ? Traverse.nextSibling : Traverse.prevSibling;
+    const focussed = Arr.find(Traverse.children(parentComp.element), Focus.hasFocus);
+    focussed.fold(Optional.none, nextNode).each((next) => Focus.focus(next as SugarElement<HTMLElement>));
+  };
+
+  const makeStepperButton = (action: VoidFunction, title: string, tooltip: string, classes: string[]) => {
+    const translatedTooltip = backstage.shared.providers.translate(tooltip);
+    return Button.sketch({
+      dom: {
+        tag: 'button',
+        attributes: {
+          'title': translatedTooltip,
+          'aria-label': translatedTooltip
+        },
+        classes: classes.concat(title)
+      },
+      components: [
+        renderIconFromPack(title, backstage.shared.providers.icons)
+      ],
+      action
+    });
+  };
+
+  return {
+    dom: {
+      tag: 'div',
+      classes: [ 'tox-number-input' ]
+    },
+    components: [
+      makeStepperButton(() => decrease(true), 'minus', 'Decrease font size', [ 'highlight-on-focus' ]),
+      memInput.asSpec(),
+      makeStepperButton(() => increase(true), 'plus', 'Increase font size', [ 'highlight-on-focus' ])
+    ],
+    behaviours: Behaviour.derive([
+      Focusing.config({}),
+      Keying.config({
+        mode: 'special',
+        onEnter: (comp) => {
+          if (Focus.hasFocus(comp.element)) {
+            memInput.getOpt(comp).each((inputWrapper) => Focus.focus(inputWrapper.element));
+            return Optional.some(true);
+          } else {
+            return Optional.none();
+          }
+        },
+        onEscape: (wrapperComp) => {
+          if (Focus.hasFocus(wrapperComp.element)) {
+            return Optional.none();
+          } else {
+            Focus.focus(wrapperComp.element);
+            return Optional.some(true);
+          }
+        },
+        onLeft: (comp) => {
+          moveFocus(comp, 'left');
```

Comment:
Is there a reason that you didn't just want to handle this by a keyboard layout that was a parent? Righting shifting focus code manually can have a few problems, because it is likely going to miss some of the finer-point details about how alloy's focus shifting code works. Specifically 

* it uses `triggerFocus` rather than focus in case the element itself has more complex focus requirements. That's not required here, because you know what you are focusing. It's more required for generalised code
* it sends a focusShifted alloy system event describing the previous and next
* it bubbles to the next layer only if it isn't handled

This code is always returning None for `<left>` and `<right>` arrow keys, and is specifically setting focus. Is there a reason that we aren't just handling it by the left and right in a toolbar group ?

Reply:
refactored changing the wrapper mode to `flow`